src/main/java/org/dom4j/rule/Rule.java

```diff
@@ -99,17 +99,17 @@ public int compareTo(Object that) {
      * @return DOCUMENT ME!
      */
     public int compareTo(Rule that) {
-        int answer = this.importPrecedence - that.importPrecedence;
+        double answer = this.importPrecedence - that.importPrecedence;
 
         if (answer == 0) {
-            answer = (int) Math.round(this.priority - that.priority);
+            answer = this.priority - that.priority;
```

Comment:
This was breaking an ordering test case as the rounding was rounding -0.5 to 0 meaning objects were being treated equally

Reply:
You may never be able to satisfy this condition, since comparing floating point numbers to exact values rarely produces expected results. You need to compare the difference between them to some predefined tolerance value.

Reply:
The test case that was breaking now works as a result of this change so I believe it does the job